gateway-js/src/index.ts

```diff
@@ -358,18 +357,23 @@ export class ApolloGateway implements GraphQLService {
   }
 
   protected async updateComposition(): Promise<void> {
-    let result: Await<ReturnType<Experimental_UpdateServiceDefinitions>>;
-    this.logger.debug('Checking service definitions...');
-    try {
-      result = await this.updateServiceDefinitions(this.config);
-    } catch (e) {
-      this.logger.error(
-        'Error checking for changes to service definitions: ' +
-          ((e && e.message) || e),
+    this.logger.debug('Checking for composition updates...');
+
+    // This may throw, but an error here is caught and logged upstream
+    const result = await this.updateServiceDefinitions(this.config);
+
+    if (isCsdlUpdate(result)) {
+      await this.updateCsdl(result);
+    } else if (isServiceDefinitionUpdate(result)) {
+      await this.updateServiceDefs(result);
+    } else {
+      throw new Error(
+        'Programming error: unexpected result type from `updateServiceDefinitions`',
       );
-      throw e;
```

Comment:
Ooh, "log an error and then rethrow it anyway" always strikes me as an odd choice so I'm happy to see it's not happening any more.